aws-xray/src/main/java/io/opentelemetry/contrib/awsxray/RemoteAttributesSpanProcessor.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.awsxray;
+
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.sdk.trace.ReadWriteSpan;
+import io.opentelemetry.sdk.trace.ReadableSpan;
+import io.opentelemetry.sdk.trace.SpanProcessor;
+
+/**
+ * RemoteAttributesSpanProcessor copies attributes aws.remote.application and aws.remote.operation
+ * from the parent span to children after receiving them. These two attributes will be used as
+ * RemoteApplication and RemoteOperation dimension values in dependency metrics generation.
+ */
+public final class RemoteAttributesSpanProcessor implements SpanProcessor {
+
+  private static final AttributeKey<String> REMOTE_APPLICATION =
+      AttributeKey.stringKey("aws.remote.application");
+  private static final AttributeKey<String> REMOTE_OPERATION =
+      AttributeKey.stringKey("aws.remote.operation");
+
+  private RemoteAttributesSpanProcessor() {}
+
+  public static RemoteAttributesSpanProcessor create() {
+    return new RemoteAttributesSpanProcessor();
+  }
+
+  @Override
+  public void onStart(Context parentContext, ReadWriteSpan span) {
+    Span parentSpan = Span.fromContextOrNull(parentContext);
+    if (!(parentSpan instanceof ReadableSpan)) {
+      return;
+    }
+    ReadableSpan parentReadableSpan = (ReadableSpan) parentSpan;
+
+    String remoteApplication = parentReadableSpan.getAttribute(REMOTE_APPLICATION);
+    if (remoteApplication != null) {
```

Comment:
Can you add test coverage for the case where `aws.remote.application` / `aws.remote.operation` are null?

Reply:
Thanks for the review. Added a new test case for null check.